### ISSUE TITLE
Mac OS X only: Blocked modal context (was: "I have no 3D view")

### ISSUE THREAD

**Issue by NWPerry** [nilsschmidt1337/ldparteditor#748]
Stacktrace:
```
[LDPartEditor RELEASE 0.8.50 @ SWT v4930 OpenGL 2.0 PLATFORM cocoa]
Unhandled event loop exception during blocked modal context.
java.lang.NullPointerException
	at org.eclipse.swt.widgets.Widget.drawRect(Widget.java:763)
	at org.eclipse.swt.widgets.Canvas.drawRect(Canvas.java:170)
	at org.eclipse.swt.widgets.Display.windowProc(Display.java:6043)
	at org.eclipse.swt.internal.cocoa.OS.objc_msgSendSuper(Native Method)
	at org.eclipse.swt.widgets.Display.applicationNextEventMatchingMask(Display.java:5284)
	at org.eclipse.swt.widgets.Display.applicationProc(Display.java:5716)
	at org.eclipse.swt.internal.cocoa.OS.objc_msgSend(Native Method)
	at org.eclipse.swt.internal.cocoa.NSApplication.nextEventMatchingMask(NSApplication.java:97)
	at org.eclipse.swt.widgets.Display.readAndDispatch(Display.java:3804)
	at org.eclipse.jface.operation.ModalContext$ModalContextThread.block(ModalContext.java:166)
	at org.eclipse.jface.operation.ModalContext.run(ModalContext.java:368)
	at org.eclipse.jface.dialogs.ProgressMonitorDialog.run(ProgressMonitorDialog.java:469)
	at org.nschmidt.ldparteditor.shells.editor3d.Editor3DWindow.run(Editor3DWindow.java:5480)
	at org.nschmidt.ldparteditor.splash.SplashScreen.run(SplashScreen.java:424)
	at org.nschmidt.ldparteditor.main.LDPartEditor.main(LDPartEditor.java:52)
```

_NWPerry wrote_: When I launch the program I have no 3D editor view. Sometimes there is a blank white panel, but not always. Error log is here: https://pastebin.com/HrdFrXpP

<img width="1392" alt="Screen Shot 2020-04-17 at 2 51 15 AM" src="https://user-images.githubusercontent.com/56901853/79540618-719ed980-8056-11ea-977f-f59ff5d5aa49.png">

[error_log.txt](https://github.com/nilsschmidt1337/ldparteditor/files/4511560/error_log.txt)

**Comment by nilsschmidt1337** [comment]
You can re-open the 3D view with the following two clicks:
<img width="1392" alt="re-open_3D_view" src="https://user-images.githubusercontent.com/11047164/79893355-3c064180-8404-11ea-998a-515e96af9bbb.png">

I can find a solution which makes it easier for the user to get a 3D view back on the screen.
Maybe I will include it in the next release.

**Comment by NWPerry** [comment]
Confirmed this is still present in 0.8.53. First time I opened the program it looked like this (after mousing over the buttons in the text editor to make them appear):
<img width="1392" alt="Screen Shot 2021-05-24 at 9 30 35 AM" src="https://user-images.githubusercontent.com/56901853/119360060-3d91e100-bc78-11eb-924d-6bbffb213520.png">

I haven't had any success with clicking the 3D button to force it to open. The fix is always to close and reopen the program 1-3 times.

The second and third time I had the empty white panel:
<img width="1392" alt="Screen Shot 2021-05-24 at 9 32 00 AM" src="https://user-images.githubusercontent.com/56901853/119360167-57332880-bc78-11eb-8b8e-3db4620ea9e8.png">

Fourth time I opened, the program behaved properly. Here's the error log file:
[error_log.txt](https://github.com/nilsschmidt1337/ldparteditor/files/6532950/error_log.txt)
